src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java

```diff
@@ -469,6 +512,13 @@ private void prepareRunStorageBucket(BatchSinkContext context) throws Exception
     if (bucket == null) {
       GCPUtils.createBucket(storage, bucketName, config.getLocation(), cmekKeyName);
     }
+    if (config.isUserManagedSchema()) {
+      GoogleCredentials googleCredentials = config.validateAndGetServiceAccountCredentials(collector);
```

Comment:
move this into prepareRun(). Fail if the asset type is BigQuery because this feature is only supported with GCS assets

Reply:
Also call this code from configurePipeline() so that validation errors can be caught at configure time

Reply:
Re: FIrst comment above: The `User Managed` toggle only shows up for GCS assets. It's always hidden for BQ assets.

Reply:
Not sure if you meant to move it to `prepareRun()` or `configurePipeline()` (or both?!). But I ended up calling it from `prepareRun()`. It swallows the `RuntimeException` when called from `configurePipeline()`.

Reply:
I meant both. 
Why?
- `configurePipeline` is called at configure time. It's easier to find and fix configuration errors while validating the pipeline in studio vs running a deployed pipeline. Note that macros are not evaluated at configure time, so macro params won't be validated here. You'll want to wrap your validation logic in if(containMacro()) checks (you can find examples of this in google-cloud repo).
- `prepareRun` is called when the pipeline is run. We want the validation logic to be present here as well because macros are resolved when prepareRun is called. So macro params can be validated here.

Reply:
> It swallows the RuntimeException when called from configurePipeline()
Don't throw RuntimeException. Add a failure to FailureCollector.

Reply:
addressed in [479534a](https://github.com/data-integrations/google-cloud/pull/1075/commits/479534aae257b81d9efe58ae4b29a15a7659560b)